src/scratchOrgFeatureDeprecation.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * Certain Org Features require a translation or should be deprecated.
+ * Encapsulates feature mappings and deprecated features.
+ */
+// P R I V A T E
+
+import { isString } from '@salesforce/ts-types';
+
+// Local
+import { Messages } from './messages';
+Messages.importMessagesDirectory(__dirname);
+const messages: Messages = Messages.loadMessages('@salesforce/core', 'scratchOrgFeatureDeprecation');
+
+const FEATURE_TYPES = {
+  // simpleFeatureMapping holds a set of direct replacement values for features.
+  simpleFeatureMapping: {
+    SALESWAVE: ['DEVELOPMENTWAVE'],
+    SERVICEWAVE: ['DEVELOPMENTWAVE'],
+  },
+  quantifiedFeatureMapping: {},
+  deprecatedFeatures: [
+    'EXPANDEDSOURCETRACKING',
+    'LISTCUSTOMSETTINGCREATION',
+    'AppNavCapabilities',
+    'EditInSubtab',
+    'OldNewRecordFlowConsole',
+    'OldNewRecordFlowStd',
+    'DesktopLayoutStandardOff',
+    'SplitViewOnStandardOff',
+    'PopOutUtilities',
+  ],
+};
+
+interface FeatureTypes {
+  simpleFeatureMapping: { [key: string]: string[] };
+  quantifiedFeatureMapping: Record<string, string>;
+  deprecatedFeatures: string[];
+}
+
+export class ScratchOrgFeatureDeprecation {
```

Comment:
this class is going to be pretty easy to test (pass various featureTypes, expect values and warnings).

I'd like to see some other very testable functions that could, for example,

1. build the scratchOrgInfo request (give inputs, get object that you'd send OR errors/warnings)
2. build the scratchOrg settings (give inputs, get deployable settings OR errors/warnings)

Then you can avoid having to mock all the server stuff since most of the complexity lies in all the possible inputs, and the nuts can handle the e2e testing.

changing this stuff to a more input/output approach can simplify the UT work